File: app/src/main/java/com/johnpetitto/orachat/ui/chatroom/ChatroomView.java
package com.johnpetitto.orachat.ui.chatroom;

import com.johnpetitto.orachat.data.chat.ChatMessage;

import java.util.List;

public interface ChatroomView {
    void showLoading(boolean show);
    void showError();
    void showInitialMessages(List<ChatMessage> messages);
    void showNewMessage(ChatMessage message);
    void showMoreMessages(List<ChatMessage> messages);
    void pageLoaded();
}

File: app/src/main/java/com/johnpetitto/orachat/ui/chatroom/ChatroomView.kt
package com.johnpetitto.orachat.ui.chatroom

import com.johnpetitto.orachat.data.chat.ChatMessage

interface ChatroomView {
    fun showLoading(show: Boolean)
    fun showError()
    fun showInitialMessages(messages: List<ChatMessage>)
    fun showNewMessage(message: ChatMessage)
    fun showMoreMessages(messages: List<ChatMessage>)
    fun pageLoaded()
}
